### ISSUE TITLE
MOGL models for roles not requiring MOGL

### ISSUE THREAD

**Issue by AA-Turner** [the-scouts/compass-interface-core#30]
In #29 `compass.core.schemas.member.MemberMOGLList` was updated to make all mandatory ongoing learning (MOGL) modules reqired. This has benefits in that we can guarantee to library consumers that these attributes will be present on the model, but does (can) lead to side effects. As a quick list (examples may have been missed):

- If a member is required by policy to have completed certain training, but this is not on his training record, we are discarding implied data that Compass needs to be updated to add said modules (i.e. the missing data reflects both no record of validation/completion, and that no learning requirement has been added). I don't see this as incredibly bad, as he would need to do the training anyhow.
- This though leads to cases where our individual's roles require (without loss of generality) M01 and GDPR, but not First Aid. Partly a philisophical question - is it this project's job to signal completion of all requirements etc, or just report what data are found?
- In a similar vein, say his only roles are President of the local group - in this case (and a few others, notably Staff roles), there is no training obligation as it is an honorary role. Do we even return MOGL data here?

cc: @rglss, @MrNoScript, @arbitrarypunter

**Comment by arbitrarypunter** [comment]
My personal opinion is that its not this projects job to be validating compliance. Other logic elsewhere should be used for that.
for part 3, i think you should continue to return the MOGL detail, and its up to whoever is looking at or parsing the data to interpret it. Also means less hassle for maintenance if the rules change

**Comment by rglss** [comment]
Agreed - the thought behind presenting the MOGL objects is sound, these are frequently used be consumers of this library for various tasks.  It isn't up to to CIC to validate which are mandatory for the role and which aren't, in an ideal situation this would be presented by Compass that a MOGL module was missing. 

However, i'm sure this would be a beneficial feature that would be re-used by a number of users, and including something would save everyone time and effort if the implementation was shared. What were you thoughts on how this could be developed and kept valid?

**Comment by AA-Turner** [comment]
Potentially via https://github.com/the-scouts/table-2 - basically a programmatic version of the R2R table. (Or we could just use the R2R table (https://github.com/the-scouts/compliance-assistant/blob/master/data/r2r_sep_20.csv) [private repo]
